bot/extensions/vote_cog.py

```diff
@@ -0,0 +1,315 @@
+			if self._seconds % 60 >= 10:
+				self._embed.set_timer_label(f'**{self._seconds // 60}:{self._seconds % 60}**')
+			else:
+				self._embed.set_timer_label(f'**{self._seconds // 60}:0{self._seconds % 60}**')
+		else:
+			if self._seconds % 60 >= 10:
+				self._embed.set_timer_label(f'**0{self._seconds // 60}:{self._seconds % 60}**')
+			else:
+				self._embed.set_timer_label(f'**0{self._seconds // 60}:0{self._seconds % 60}**')
+
+		self._embed.build()
+		await self._msg.edit(embed=self._embed)
+
+
+# Limitation: It's only possible to run 1 poll at a time. Well you can run 2 or more but it's not gonna be asynchronous
+# Suggestion: Probably use (if exists) discord's built-in threads, to separate the processes.
+class PollCog(Cog):
+	def __init__(self, bot: discord.ext.commands.Bot) -> None:
+		self.bot = bot
+
+	def make_sequence(self, seq):
+	    if seq is None:
+	        return ()
+	    if isinstance(seq, Sequence) and not isinstance(seq, str):
+	        return seq
+	    else:
+	        return (seq,)    
+
+	def message_check(self, channel=None, author=None, content=None, ignore_bot=True, lower=True) -> Callable[[discord.Message], bool]:
+	    channel = self.make_sequence(channel)
+	    author = self.make_sequence(author)
+	    content = self.make_sequence(content)
+	    if lower:
+	        content = tuple(c.lower() for c in content)
+	    def check(message: discord.Message):
+	        if ignore_bot and message.author.bot:
+	            return False
+	        if channel and message.channel not in channel:
+	            return False
+	        if author and message.author not in author:
+	            return False
+	        actual_content = message.content.lower() if lower else message.content
+	        if content and actual_content not in content:
+	            return False
+	        return True
+	    return check
+
+	@hybrid_group(name="poll", help="Poll commands")
+	async def poll_group(self, ctx) -> None:
+		if ctx.invoked_subcommand is None:
+		    await send_command_help(ctx)
+
+	@poll_group.command(name='create', help='Create a poll')
+	async def vote(self, ctx: Context, *, title: str, options_count: int = 2, poll_time: int = 120) -> None:
+		if options_count < 2:
+			await ctx.interaction.response.send_message('Only 2 or more options is allowed.', ephemeral=True)
+			return
+		cancel_keywords = ['!abort']
+
+		# if not title:
+		# 	# Create title embed
+		# 	title_embed = Embed(
+		# 		color=self.bot.default_color,
+		# 		title='POLL CREATION INFO',
+		# 		description='**Input poll title**'
+		# 	)
+		# 	# Send a DM to the command executor
+		# 	await ctx.author.send(embed=title_embed)
+
+		# 	# TODO: Think if input title can be invalid
+		# 	title = await self.bot.wait_for('message', check=self.message_check(ctx.author.dm_channel))
+
+		# 	# Check if user cancelled the poll
+		# 	if title.content.lower().strip() in cancel_keywords:
+		# 		await ctx.interaction.response.send_message('Aborted', ephemeral=True)
+		# 		return
```

Comment:
I see a lot of commented code. Is there a particular reason for this, or could it be removed?

Reply:
Yes, the one you showed I think I might remove it. Thanks for your remark.